zk-token-sdk/src/encryption/auth_encryption.rs

```diff
@@ -172,10 +174,14 @@ impl EncodableKey for AeKey {
 impl SeedDerivable for AeKey {
     fn from_seed(seed: &[u8]) -> Result<Self, Box<dyn error::Error>> {
         const MINIMUM_SEED_LEN: usize = AE_KEY_LEN;
+        const MAXIMUM_SEED_LEN: usize = 65535;
 
         if seed.len() < MINIMUM_SEED_LEN {
             return Err(AuthenticatedEncryptionError::SeedLengthTooShort.into());
         }
+        if seed.len() > MAXIMUM_SEED_LEN {
+            return Err(AuthenticatedEncryptionError::SeedLengthTooLarge.into());
```

Comment:
Oh sorry, you'll also need this!
```suggestion
            return Err(AuthenticatedEncryptionError::SeedLengthTooLong.into());
```